site/src/pages/tutorials/thrift/blog/implement-create.mdx

```diff
@@ -0,0 +1,155 @@
+In this step, we'll fill in one of the service methods to create a blog post and again, try making a call to a service method.
+
+<TutorialSteps current={4} />
+
+## What you need
+
+You need to have the following obtained from previous steps:
+
+- [Java code generated](/tutorials/thrift/blog/create-service#2-compile-the-thrift-file)
+- Main.java
+- BlogServiceImpl.java
+- BlogClient.java
+
+## 1. Implement server-side
+
+Let's implement a service method to create a blog post.
+
+1. In the `BlogServiceImpl` class, create an ID generator to issue temporary blog post IDs and a map to contain blog posts.
+  ```java filename=BlogServiceImpl.java
+  import java.util.Map;
+  import java.util.concurrent.ConcurrentHashMap;
+  import java.util.concurrent.atomic.AtomicInteger;
+
+  public class BlogServiceImpl implements BlogService.AsyncIface {
+      private final AtomicInteger idGenerator = new AtomicInteger();
+      private final Map<Integer, BlogPost> blogPosts = new ConcurrentHashMap<>();
+      ...
+  }
+  ```
+2. In the `createBlogPost()` method, create a `BlogPost` object with a generated ID and request parameters.
+  ```java filename=BlogServiceImpl.java
+  import java.time.Instant;
+  ...
+  @Override
+  public void createBlogPost(CreateBlogPostRequest request, AsyncMethodCallback<BlogPost> resultHandler)
+          throws TException {
+      final int id = idGenerator.getAndIncrement();
+      final Instant now = Instant.now();
+      final BlogPost blogPost = new BlogPost()
+              .setId(id)
+              .setTitle(request.getTitle())
+              .setContent(request.getContent())
+              .setModifiedAt(now.toEpochMilli())
+              .setCreatedAt(now.toEpochMilli());
+  }
+  ```
+2. In the `createBlogPost()` method, save the post information in the `blogPosts` map and return the information of the blog post created to the `resultHandler`.
+  ```java filename=BlogServiceImpl.java
+  @Override
+  public void createBlogPost(CreateBlogPostRequest request, AsyncMethodCallback<BlogPost> resultHandler)
+          throws TException {
+      ...
+      blogPosts.put(id, blogPost);
+      final BlogPost stored = blogPost;
+      resultHandler.onComplete(stored);
+  }
+  ```
+
+## 2. Implement client-side
+
+In the `BlogClient` class, add a method to send a request to create a blog post.
+
+```java filename=BlogClient.java
+import example.armeria.blog.thrift.BlogPost;
+import example.armeria.blog.thrift.CreateBlogPostRequest;
+...
+BlogPost createBlogPost(String title, String content) throws TException {
+    final CreateBlogPostRequest request =
+            new CreateBlogPostRequest().setTitle(title)
+                                       .setContent(content);
+    return blogService.createBlogPost(request);
+}
+```
+
+## 3. Test creating a blog post
+
+Now, let's write the test code to test what we implemented so far in this step.
+Note the code is minimal to keep things simple.
+
```

Comment:
How about adding them in the `build.gradle` code snippet at `index.mdx`?

Reply:
I updated the `build.gradle`: [ae779aa](https://github.com/line/armeria/pull/4865/commits/ae779aad118c6e3239949e05f3902735724bf9f2)

Please let me know if there is any coding convention issue or anything (_ _)

Reply:
Also, I get an error when I try to build with this `build.gradle` because of the Armeria version this variable(`${versions['org.junit:junit-bom']}`) gives when I render the site locally. Simply this version does not seem to exist.

```
Could not find com.linecorp.armeria:armeria:1.22.2-SNAPSHOT.
        Searched in the following locations:
          - https://repo.maven.apache.org/maven2/com/linecorp/armeria/armeria/1.22.2-SNAPSHOT/maven-metadata.xml
          - https://repo.maven.apache.org/maven2/com/linecorp/armeria/armeria/1.22.2-SNAPSHOT/armeria-1.22.2-SNAPSHOT.pom
```

I'm not sure if I am missing something. I just wanted to check if it's okay as it is 🙂

Reply:
> Simply this version does not seem to exist.

We're having trouble uploading snapshot versions at the moment 😅  (due to the following CI failure https://github.com/line/armeria/actions/runs/5010142242/jobs/8979748543)
We're working on fixing this issue.

Meanwhile, if we modify the version (e.g. `1.22.2-SNAPSHOT` ->  `1.22.1`) I guess the build will work as expected.